mail/src/main/java/org/jboss/as/mail/extension/MailSessionAdd.java

```diff
@@ -90,19 +90,19 @@ protected void performRuntime(OperationContext context, ModelNode operation, Mod
         installRuntimeServices(context, address, fullTree);
     }
 
-    private static void addCredentialStoreReference(ServerConfig serverConfig, OperationContext context, ModelNode model, ServiceBuilder serviceBuilder, String... modelFilter) throws OperationFailedException {
+    private static void addCredentialStoreReference(ServerConfig serverConfig, OperationContext context, ModelNode model, ServiceBuilder serviceBuilder, final String modelFilter1, final String modelFilter2) throws OperationFailedException {
         if (serverConfig != null) {
             ModelNode filteredModelNode = model;
-            if (modelFilter != null && modelFilter.length > 0) {
-                for (String path : modelFilter) {
-                    if (filteredModelNode.get(path).isDefined())
-                        filteredModelNode = filteredModelNode.get(path);
-                    else
-                        break;
-                }
+
+            if (filteredModelNode.get(modelFilter1).isDefined()
+                    && filteredModelNode.get(modelFilter2).isDefined()) {
+                filteredModelNode = filteredModelNode.get(modelFilter2);
```

Comment:
Notice `filteredModelNode` is being reassigned on each for loop; it is traversing the model assigned in L95 following a path.

The equivalent `if` to replace the for loop would be:

```
if (filteredModelNode.get(modelFilter1).isDefined()
        && filteredModelNode.get(modelFilter1).get(modelFilter2).isDefined()) {
    filteredModelNode = filteredModelNode.get(modelFilter1).get(modelFilter2);
}
```

Reply:
Two DMR usage notes:

1) node.get(x).isDefined() is the same as node.hasDefined(x) except it doesn't create an undefined node (which is good)

2) get and hasDefined support varargs, and hasDefined will return false as soon as it can.

So

```
if (filteredModelNode.hasDefined(modelFilter1, modelFilter2)) {
    filteredModelNode = filteredModelNode.get(modelFilter1, modelFilter2);
}
```

Reply:
@yersan @bstansberry thanks for your feedback. I have changed here according to it.